### PR TITLE
desktop: #1774 add file dialogs when the Ruffle desktop app is opened

### PR DESCRIPTION
**Pull request by SamMorrowDrums**
So I think this PR is ready and it's been tested on Linux and windows (a Mac user test would also be appreciated). I have one reservation as the published crate https://github.com/saurvs/nfd-rs is stale, so I have referenced a forked branch with open PR instead.

Notes: 

I'm on Ubuntu/Linux and I needed `libgtk-3-dev` in order to install `nfd-rs`

I also get a console warning
```
Gtk-Message: 16:35:11.230: GtkDialog mapped without a transient parent. This is discouraged.

```

### CODE REVIEW

**Comment by SamMorrowDrums** [comment]
Linux tests are failing because of  missing`libgtk-3-dev` as mentioned in my description.

File: desktop/src/main.rs [hunk 1]
```diff
@@ -128,17 +129,39 @@ fn load_movie_from_path(
...
+                Url::from_file_path(absolute_path)
+                    .map_err(|_| "Path must be absolute and cannot be a URL")?
+            } else {
+                Url::parse(path.to_str().unwrap_or_default())
+                    .map_err(|_| "Input path is not a file and could not be parsed as a URL.")?
+            }
+        }
+        None => {
+            let result = open_file_dialog(Option::from("swf"), None).unwrap_or_else(|e| {
+                panic!(e);
```
Comment (Herschel) [inline desktop/src/main.rs:145]: Let's not panic here and instead return the error upward with `?`. (`map_err` if you need to turn it into a different type of error).

File: desktop/Cargo.toml [hunk 2]
```diff
@@ -24,6 +24,7 @@ url = "2.2.0"
 clipboard = "0.5.0"
 dirs = "3.0"
 isahc = "0.9.13"
+nfd = {git = "https://github.com/EmbarkStudios/nfd-rs", ref="90ff6d2"}
```
Comment (Herschel) [inline desktop/Cargo.toml:27]: `rev` not `ref`

**Comment by Herschel** [comment]
Thanks and welcome! The functionality seems great, the only question is if `nfd` is the crate we want to use. The lack of updates is concerning, additionally https://github.com/hecrj/iced/issues/317 (iced is a very likely candidate for our full GUI).

Another possibility is https://github.com/jdm/tinyfiledialogs-rs/ -- could you maybe give that a shot and see how it compares? https://github.com/jdm/tinyfiledialogs-rs/issues/19 is a little concerning, but the crate seems maintained at least, and maybe a PR could be made to update it to the latest.

**Comment by SamMorrowDrums** [comment]
Thanks for feedback, I'll have a look tomorrow. I would probably take a maintained crate with known flaws (within reason) over an abandoned one any day of the week! If it works well, could be better alternative. I'll give iced a look too.

**Comment by SamMorrowDrums** [comment]
@Herschel I've added #1841 a tiny file dialogs PR (including bumped version of the C library). So now on Linux for me they both share the same behaviour - if they work x-platform (as both libraries claim), then it's really a matter of your preference I guess, but neither should be a blocker, so if the alternative is expected to work better with `iced` then I guess go with that one.

**Comment by SamMorrowDrums** [comment]
One thing I will say that is going for #1841 is that it bakes in it's C dependency, so doesn't require additional OS package for building on Linux.

**Comment by SamMorrowDrums** [comment]
One thing I will say that is going for #1841 is that it bakes in it's C dependency, so doesn't require additional OS package for building on Linux.

**Comment by Herschel** [comment]
Closed in favor of #1841.

**Pull request closed without merging**